### PR TITLE
[WIP] Track job failures in a structured way.

### PR DESCRIPTION
**Pull request by jmchilton**
Still working on this - built from @bernt-matthias's branch https://github.com/galaxyproject/galaxy/pull/6338.

### CODE REVIEW

**Comment by jmchilton** [comment]
@bernt-matthias So this breaks all the new tests in error_detect.xml. Looking over those tests, ``failure_reasons`` might not be a great name for this property I added either because there are info level messages.

Should I should maybe this redo this and call the property ``runtime_messages`` or ``job_messages`` and those tests need to be modified since we are no longer modifying the programs stdout or stderr - which I still think is a good thing. I'll think about this some more, sorry for mucking things up.

**Comment by bernt-matthias** [comment]
Fine with any of the suggested names. 

Regarding the test: currently the fatal_oom regex is tested only in conjunction with fatal(_oom) exit codes. But we should also test it alone, i.e. without an exit code. 

I'm a bit lost in the tests, can I see somewhere the the reported levels for each of the tests.

File: lib/galaxy/jobs/output_checker.py [hunk 1]
```diff
@@ -88,32 +114,16 @@ def check_output(tool, stdout, stderr, tool_exit_code, job):
                     #   - If it matched, then determine the error level.
                     #       o If it was fatal, then we're done - break.
                     # Repeat the stdout stuff for stderr.
-                    # TODO: Collapse this into a single function.
+                    # TODO could test for stderr first? Reason: I would expect it to be smaller and contain the errors
```
Comment (bernt-matthias) [inline lib/galaxy/jobs/output_checker.py]: I agree. lets just reverse order

**Comment by bernt-matthias** [comment]
Hi @jmchilton any progress on this. Just let me know if there is anything I can help.

File: lib/galaxy/jobs/output_checker.py (hunk 1 above)
Comment (jmchilton) [reply lib/galaxy/jobs/output_checker.py]: will do

**Comment by jmchilton** [comment]
Rebased and took a crack at polishing this.

Barring test failures I think what is left is to restore/improve the UX around reporting why jobs failed.

**Comment by jmchilton** [comment]
Visual of job messages no longer being pre-pending to job stdout/stderr - now appear as structured data in the show parameters UI:

screen shot 2018-12-11 at 9 14 00 am

**Comment by jmchilton** [comment]
Closing in lieu of #7095

**Pull request closed without merging**